File: javascript/matrix/matrix.js
export class Matrix {
  constructor(matrix_string) {
    this.rows = matrix_string.split('\n').map(row => row.split(" ").map(element => +element))
    this.columns = this.rows[0].map((_, index) => this.rows.map(row => row[index]))   
  }
}

File: ruby/matrix/matrix.rb
class Matrix
  attr_reader :rows
  attr_reader :columns

  def initialize(matrix_string)
    @rows = extract_rows(matrix_string)
    @columns = rows.transpose
  end

  private def extract_rows(matrix_string)
    matrix_string.each_line.map { |row| row.split.map(&:to_i) }
  end
end
